src/jit/lowerxarch.cpp

```diff
@@ -2576,9 +2576,19 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
     GenTree* op2 = node->gtGetOp2();
     GenTree* op3 = nullptr;
 
+    if (HWIntrinsicInfo::lookupCategory(intrinsicId) == HW_Category_IMM)
```

Comment:
Why was this moved up?

Reply:
Before this PR, IMM intrinsics do not have NoContainment ones. But AVX2 Gather* are IMM intrinsic AND memory-load intrinsic that cannot work with containment (so they have NoContainment flag).

So we need to move the immediate containment above the early exiting, which marks the IMM argument of Gather before return.

Reply:
Still not sure I see the problem.

There are multiple intrinsics which can be sometimes contained, or partially contained (just the immediate node). You don't mark those as `NoContainment`, but exit below instead.